net/redistest/redistest.go

```diff
@@ -2,57 +2,71 @@ package redistest
 
 import (
 	"context"
-	"net"
-	"os/exec"
-	"strconv"
 	"testing"
 	"time"
 
 	"github.com/go-redis/redis/v8"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
 )
 
 func NewTestRedis(t *testing.T) (address string, done func()) {
 	return NewTestRedisWithPassword(t, "")
 }
 
 func NewTestRedisWithPassword(t *testing.T, password string) (address string, done func()) {
-	l, err := net.Listen("tcp", "127.0.0.1:0")
-	if err != nil {
-		t.Fatal(err)
-	}
-	address = l.Addr().String()
-	port := strconv.Itoa(l.Addr().(*net.TCPAddr).Port)
-	l.Close()
-
-	args := []string{"--port", port}
+	var args []string
 	if password != "" {
 		args = append(args, "--requirepass", password)
 	}
 
-	ctx, stop := context.WithCancel(context.Background())
+	start := time.Now()
+
+	// first testcontainer start takes longer than subsequent
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
 
-	cmd := exec.CommandContext(ctx, "redis-server", args...) // #nosec
-	if err := cmd.Start(); err != nil {
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image:        "redis:6-alpine",
+			Cmd:          args,
+			ExposedPorts: []string{"6379/tcp"},
```

Comment:
Does a test container gets it's own IP?

Reply:
It gets a mapped port on localhost that we obtain using .Endpoint below

Reply:
sounds like we could test sharding with this too, but we need to change the port for the next ones :)